public/uploadStudentList.php

```diff
@@ -0,0 +1,112 @@
+
+<?php 
+require_once "../modules/models/user.php";
+require_once "../common.php";
+
+    if (isset($_POST['submit2'])) 
+	{
+        
+		$students = fopen('studentList.csv', 'r');
+        $count = 0;
+		while (($studentData = fgetcsv($students, 1000, ",")) !== FALSE) 
+        {
+                $user = new User();
+                $user->student_id = $studentData[0];
+                $user->first_name = $studentData[1];
+                $user->last_name = $studentData[2];
+                $user->email = $studentData[3];
+                $user->is_admin = 0;
+                $user->is_instructor = 0;
+                $user->password_digest = password_hash("welcome", PASSWORD_DEFAULT);
+    
+                try {
+                    $user->save();
+                    $count++;
+                    $create_success = true;
+                }  catch(PDOException $error) {
+                     echo "<br>" . $error->getMessage();
+                }
+            
+    
+        }
+        
+        
+        fclose($students);
+        $filePath = 'studentList.csv';
+        unlink($filePath);
+
+	}
+?>
+
+<?php include "templates/header.php"; ?>
+
+<?php
+    if(isset($_POST['submit2']) && $create_success){
+        echo "<h3> $count Students added successfully</h3><br>";
+    }
+?>
+
+<form enctype='multipart/form-data' action='' method='post'>
+
+    <h2>
+        <label>Upload Student List</label> 
+    </h2>
+    <br>
+    <label for="filename">Select File</label>
+    <input size='50' type='file' name='filename' accept=".csv">
+    </br>
+
+    <input type='submit' name='submit' value='Upload List'>
+ 
+</form>
+
+
+<?php
+    if(isset($_POST["submit"])){
+
+        global $studentData;
+        $studentData = array();
+        $row=0;
+
+        echo "<h5>Users to be added </h5>";
+
+        $handle = fopen($_FILES['filename']['tmp_name'], "r");
+		$headers = fgetcsv($handle, 1000, ",");
+        while (($data = fgetcsv($handle, 1000, ",")) !== FALSE) 
+		{
+            $studentData[$row] = array();
+
+            $num = count($data);
+            for ($c=0; $c<$num; $c++){
+                echo $data[$c] . ", ";
+                $studentData[$row][$c] = $data[$c];
+            }
+            echo "<br>";
+            
+            $row++;
+            
+        }
+        fclose($handle);
+
+        $fp = fopen('studentList.csv', 'w');
+        foreach($studentData as $fields){
+            fputcsv($fp, $fields);
```

Comment:
for the sake of doing it right I think it'd be better to include like a timestamp or something like that in the file name, otherwise if two people try to do this at the same time its gonna be a problem

Reply:
The way I understood it, the temp file gets deleted at the end of the read of the php file, and we need to have the info for a second read, so it needs to be  a different file (unless I misunderstood how this works, which is very possible) 

I'll look into adding a timeStamp in the file name

Reply:
renaming is the same as moving, sooo if you move the tmp file away and give it a new name it will no longer be deleted, there's this function called `move_uploaded_file`